### DISCUSSION TITLE
Detecting secrets in Dockerfiles when using misconfiguration mode

### DISCUSSION THREAD

**Discussion by huornlmj** [aquasecurity/trivy#6582] [Ideas]
### Description

Consider the following example Dockerfile which contains security issues from https://github.com/madhuakula/docker-security-checker/blob/master/Dockerfile
```Dockerfile
FROM ubuntu:latest
LABEL MAINTAINER "Madhu Akula"

ENV SECRET AKIGG23244GN2344GHG
ENV GITLAB_API_ID gig32oig3bgi34gb43gb43uigb43i 

WORKDIR /app

ADD app /app
COPY README.md /app/README.md
ADD code /tmp/code
RUN sudo apt-get udpate

RUN apt-get update && apt-get install -y htop

CMD ["/bin/bash", "/app/entrypoint.sh"]
```
If you scan this with `$ trivy config Dockerfile` it will detect many issues, but it will not detect any instances of potential secrets in the file.  This does not appear to be possible with the `filesystem` mode either, in that `$ trivy fs Dockerfile` will ignore the Dockerfile.

How do we use Trivy to detect secrets in Dockerfiles?

### Target

Filesystem

### Scanner

Secret

**Comment by simar7** [comment]
As per the docs [here](https://aquasecurity.github.io/trivy/v0.50/docs/coverage/iac/docker/#secret) it should just work out of the box.

As for the secrets you have, I believe only the GitLab PAT is supported for now as seen [here](https://github.com/aquasecurity/trivy/blob/29b8faf5faaa02e463cbb54465563b40d5667bf4/pkg/fanal/secret/builtin-rules.go#L154-L161) but you could add a custom rule for the secret that you are looking for [here](https://aquasecurity.github.io/trivy/v0.50/docs/scanner/secret/#custom-rules).

@DmitriyLewen

**Reply by itaysk** [reply]
@simar7 is there a reason why we wouldn't add support for this type of secret?

**Reply by simar7** [reply]
AFAIK there's nothing special about it, we just don't support it today. I can create this into a feature request.

Correct me if I'm wrong @knqyf263 as you're more familiar with secret scanning.

**Reply by knqyf263** [reply]
> AFAIK there's nothing special about it, we just don't support it today.

Right. Our secret rules are mainly driven by the community. No one has requested GitLab API ID.
